packages/firebase_crashlytics/firebase_crashlytics/example/lib/main.dart

```diff
@@ -0,0 +1,213 @@
+              ));
+
+              // Delay crash for 5 seconds
+              sleep(const Duration(seconds: 5));
+
+              // Use FirebaseCrashlytics to throw an error. Use this for
+              // confirmation that errors are being correctly reported.
+              FirebaseCrashlytics.instance.crash();
+            }),
+        RaisedButton(
+            child: const Text('Throw Error'),
+            onPressed: () {
+              _scaffoldKey.currentState.showSnackBar(SnackBar(
+                content: Text('Thrown error has been caught \n'
+                    'Please crash and reopen to send data to Crashlytics'),
+                duration: Duration(seconds: 5),
+              ));
+
+              // Example of thrown error, it will be caught and sent to
+              // Crashlytics.
+              throw StateError('Uncaught error thrown by app');
+            }),
+        RaisedButton(
+            child: const Text('Async out of bounds'),
+            onPressed: () {
+              _scaffoldKey.currentState.showSnackBar(SnackBar(
+                content: Text(
+                    'Uncaught Exception that is handled by second parameter of runZonedGuarded \n'
+                    'Please crash and reopen to send data to Crashlytics'),
+                duration: Duration(seconds: 5),
+              ));
+
+              // Example of an exception that does not get caught
+              // by `FlutterError.onError` but is caught by
+              // `runZonedGuarded`.
+              runZonedGuarded(() {
+                Future<void>.delayed(const Duration(seconds: 2), () {
+                  final List<int> list = <int>[];
+                  print(list[100]);
+                });
+              }, FirebaseCrashlytics.instance.recordError);
+            }),
+        RaisedButton(
+            child: const Text('Record Error'),
+            onPressed: () async {
+              try {
+                _scaffoldKey.currentState.showSnackBar(SnackBar(
+                  content: Text('Recorded Error  \n'
+                      'Please crash and reopen to send data to Crashlytics'),
+                  duration: Duration(seconds: 5),
+                ));
+                throw 'error_example';
+              } catch (e, s) {
+                // "context" will append the word "thrown" in the
+                // Crashlytics console.
+                await FirebaseCrashlytics.instance
+                    .recordError(e, s, context: 'as an example');
+              }
+            }),
+      ],
+    );
+  }
+
+  /// Given a [Widget], wrap and return a [MaterialApp].
+  MaterialApp withMaterialApp(Widget body) {
+    return MaterialApp(
+      home: Scaffold(
+        key: _scaffoldKey,
+        appBar: AppBar(
+          title: const Text('Crashlytics example app'),
+        ),
+        body: body,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return withMaterialApp(Center(child: appContent));
```

Comment:
Preference is to split build logic into smaller widgets as opposed to methods (see https://iiro.dev/2018/12/11/splitting-widgets-to-methods-performance-antipattern/) so may want to consider updating the code here